### PR TITLE
modify yolov3 loss to get better detect result

### PR DESCRIPTION
**Pull request by w19787**
i met issue that nothing or very limited objects were detected with fine tune weights, after 2 or 80 epochs. since i used coco 2014 dataset, I made a big mistake that the box value of 2014 is: y1, x1, y2, x2. After change to x1, y1, x2, y2, the major problem fixed.

but i was confused that why only limited objects were detected even the box value is misaligned after 80 epochs. and part of loss code also confuse me. So i make some changes on yolov3 tf2 project. it seems to me that it work better. Hope it can help other members to solve their similar issue. 

The result is tested on 5th epoch with lr(1e-4), batch_size(16):
Origin result:
image

New result:
image

You can notice that the confidence level of new result is much better than origin result.
if anything i misunderstand, please correct me. thanks

### CODE REVIEW

**Comment by zzh8829** [comment]
Hi this is interesting change. The current implementation is true to yolov3 paper. Could you separate your custom version out as a different optional loss function? I am happy with including different implementations but i think the default one shouldn't have custom optimizations

**Comment by w19787** [comment]
The giou might not be necessary, but i do have two questions. hope you can help me to understand.
Any reason we should not use with_logit=True when we use cross_entropy loss? The differentials of cross_entropy functions without sigmoid might create unstable result.
image

And why not to use pred_box instead pred_xywh for box loss? it seems that pred_box is easier to understand and also reduce loss computation.

thanks.

**Comment by maghsk** [comment]
I think we can not use GIoU only for choosing  the best IoU.
Could we use GIoU directly as a loss function?

**Comment by bradbradlin** [comment]
@w19787 thanks. The amended loss works well for my use case too.
